bootstrap/kubeadm/api/v1beta1/kubeadmconfig_webhook_test.go

```diff
@@ -310,3 +310,18 @@ func TestClusterValidate(t *testing.T) {
 		})
 	}
 }
+
+func TestKubeadmConfigDefault(t *testing.T) {
+	g := NewWithT(t)
+
+	kc := &KubeadmConfig{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "foo",
+			Namespace: metav1.NamespaceDefault,
+		},
+	}
```

Comment:
What about changing this in a table test so we verify both that default is applied if the values is not set and also if other values are preserved if provided.

Also, let's make sure to call

t.Run("for KubeadmConfig", utildefaulting.DefaultValidateTest(kc))

So we establish a signal that validation and defaulting do not conflict each other.
Same for KubeadmConfigTemplate

Reply:
Thanks for the hints. I just noticed that tests are actually failing. I'm still not very familiar with the conventions.